code/parse/sexp.h

```diff
@@ -925,6 +943,11 @@ const char *CTEXT(int n);
 // flags for sexpressions -- masked onto the end of the type field
 #define SEXP_FLAG_PERSISTENT				(1<<31)		// should this sexp node be persistant across missions
 #define SEXP_FLAG_VARIABLE					(1<<30)
+#define SEXP_FLAG_LIST_CONTAINER			(1<<29)
+#define SEXP_FLAG_MAP_CONTAINER				(1<<28)
+#define SEXP_FLAG_CONTAINER_MODIFIER		(1<<27)
```

Comment:
Do note that a few lines down, bits 29 through 27 are already in use by `SEXP_VARIABLE_SAVE_ON_MISSION_PROGRESS` etc.  You'll need to use bits 26 through 24.

This is not documented too well, but the sexp node type, the sexp variable type, and these extra flags are all stuffed into the sexp type field.  The problem is, different flags are used in different fields.  Some of them are used together and some aren't.

Reply:
Ah, so the `SEXP_VARIABLE_*` persistence flags are actually SEXP flags?

Also, I don't think we really need separate SEXP flags for list and map containers, especially given how few open slots are available for new SEXP flags.

If the code needs to know the type of a container associated with a `sexp_node`, it can use
```
int index = get_sexp_container_index(Sexp_nodes[node].text);
Assert(index != -1);
auto &container = Sexp_containers[index];
```
and then call `container.is_map()` or `container.is_list()`.

Reply:
On second thought, since we already have `SEXP_ATOM_CONTAINER`, `SEXP_FLAG_CONTAINER` would just be redundant, so I'll get rid of that, too. We'll keep `SEXP_FLAG_CONTAINER_MODIFIER`, though.